### PR TITLE
zig: rewrite

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Motivation for this change

substituteAll is not used anymore.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/zig/default.nix [hunk 1]
```diff
@@ -33,12 +43,12 @@ llvmPackages.stdenv.mkDerivation rec {
...
+    homepage = "https://ziglang.org/";
     description =
       "General-purpose programming language and toolchain for maintaining robust, optimal, and reusable software";
-    homepage = "https://ziglang.org/";
     license = licenses.mit;
+    maintainers = with maintainers; [ andrewrk AndersonTorres ];
     platforms = platforms.unix;
-    maintainers = [ maintainers.andrewrk ];
     # See https://github.com/NixOS/nixpkgs/issues/86299
     broken = stdenv.isDarwin;
```
Comment (AndersonTorres) [inline pkgs/development/compilers/zig/default.nix:53]: Can it be done via llvmPackages.stdenv?

Comment (Mic92) [reply pkgs/development/compilers/zig/default.nix:53]: I don't understand your question. The problem here is that the libc used in nixpkgs on MacOS is too old.

Comment (AndersonTorres) [reply pkgs/development/compilers/zig/default.nix:53]: @Mic92 

Specifically, `broken = llvmPackages.stdenv.isDarwin;`

Comment (Mic92) [reply pkgs/development/compilers/zig/default.nix:53]: Should not make a difference actually.

**Pull request merged**